cardshowviewtakenpicturesview/src/main/java/br/com/stant/libraries/cardshowviewtakenpicturesview/camera/CameraFragment.java

```diff
@@ -537,51 +618,108 @@ public void setPhotos(ArrayList<CameraPhoto> photos) {
 
     @Override
     public void takePicture() {
-        PhotoResult photoResult = mCameraSetup.getFotoapparat().takePicture();
+        PhotoResult photoResult = mFotoapparat.takePicture();
         String uuid             = UUID.randomUUID().toString();
         File photoPath          = new File(mPath.toString() + "/" + JPG_FILE_PREFIX + uuid + JPG_FILE_SUFFIX);
 
-        mDialogLoader.showLocalLoader();
-
-        photoResult.saveToFile(photoPath);
+        disableCaptureButton();
 
-        photoResult.toBitmap().whenDone(
-                (bitmapPhoto) -> {
-                    Bitmap bitmap = bitmapPhoto.bitmap;
-                    Integer rotationDegrees = bitmapPhoto.rotationDegrees;
-                    if (mSaveMode.getMode().equalsIgnoreCase(SAVE_ONLY_MODE)) {
-                        mImageGenerator.scaleAndSaveInPictures(bitmap, rotationDegrees, UUID.randomUUID().toString());
-                        mDialogLoader.hideLocalLoader();
-                    } else {
-                        mImageGenerator.generateCardShowTakenImageFromCamera(bitmap, getLensPosition(), rotationDegrees,
-                                new CardShowTakenCompressedCallback() {
+        if (saveModeOnlyNotSelected()) {
+            showSaveImageLoader();
+        }
 
-                                    @Override
-                                    public void onSuccess(Bitmap bitmap, String imageFilename, String tempImagePath) {
-                                        CameraPhoto cameraPhoto = new CameraPhoto(imageFilename,
-                                                tempImagePath, new Date(), new Date());
+        final Disposable subscribe = photoResult.toBitmap()
+                .adapt(SingleAdapter.toSingle())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        (bitmapPhoto) -> {
+                            Bitmap bitmap = bitmapPhoto.bitmap;
+                            Integer rotationDegrees = bitmapPhoto.rotationDegrees;
+                            if (saveModeOnlySelected()) {
+                                mImageGenerator.scaleAndSaveInPictures(bitmap, rotationDegrees, UUID.randomUUID().toString());
+                                enableCaptureButton();
+                            } else {
+                                mImageGenerator.generateCardShowTakenImageFromCamera(bitmap, mCameraSetup.getLensPosition(), rotationDegrees,
+                                        new CardShowTakenCompressedCallback() {
```

Comment:
the listener let the method very large, you can create a method to get separate ?

Reply:
Separeted the function on a private method called "generateCardShowTakenImageFromCamera"